addons/sale/models/sale_order.py

```diff
@@ -1295,6 +1295,24 @@ def _get_invoiceable_lines(self, final=False):
 
         return self.env['sale.order.line'].browse(invoiceable_line_ids + down_payment_line_ids)
 
+    def process_zero_value_orders(self):
+        """
+        Processes sales orders when there is no payment transaction and zero 
+        amount total and creates invoices if necessary.
+
+        :return: True if invoice is generated automatically, False otherwise
+        :rtype: bool
+        """
+        self.action_confirm()
+        auto_invoice = self.env['ir.config_parameter'].get_param('sale.automatic_invoice')
+        if not auto_invoice:
+            return False
+        
+        if not self.amount_to_invoice and not self.amount_paid and self.reward_amount:
```

Comment:
```suggestion
        if self.reward_amount:
```
Isn't this condition is enough :thinking: